src/cartridges/int_adyen_overlay/cartridge/scripts/adyenCheckout.js

```diff
@@ -196,20 +196,14 @@ function doPaymentsCall(order, paymentInstrument, paymentRequest) {
       }
       paymentResponse.decision = 'ACCEPT';
       paymentResponse.threeDS2 = true;
-      let token3ds2;
-      if (responseObject.authentication['threeds2.fingerprintToken']) {
-        token3ds2 = responseObject.authentication['threeds2.fingerprintToken'];
-      } else if (responseObject.authentication['threeds2.challengeToken']) {
-        token3ds2 = responseObject.authentication['threeds2.challengeToken'];
-      }
-      paymentResponse.token3ds2 = token3ds2;
+
+      paymentResponse.token3ds2 = responseObject.action.token;
```

Comment:
Is the `token3ds2` still used later? In case the action is already on the frontend I would suggest to let the component handle the `action` instead of "manually" handling the tokens with the threeDS2 components. 

I guess you have another ticket to actually do what I just suggested so if the case is that you want to tackle this in a separate PR pls ignore this comment :)

Reply:
This token is indeed used later  in the redirect, for which we still use a form post for now rather than the action. 
Indeed the story which is now on hold does away with this and will use the action rather than parse these kinds of fields manually (for both 3ds2 as well as 3ds1, redirect, etc.

Reply:
thanks for the elaborate answer! 👍